### PR TITLE
Comment widget Course detail page

### PR DESCRIPTION
**Pull request by awais786**
Implement comments widget on course-detail page.

https://openedx.atlassian.net/browse/ECOM-6042

### CODE REVIEW

**Comment by awais786** [comment]
screen shot 2017-02-09 at 11 52 42 pm

**Comment by awais786** [comment]
@bderusha Please review this PR. Its for comments widget on course detail page.

**Comment by awais786** [comment]
screen shot 2017-02-10 at 3 58 37 pm

File: course_discovery/apps/publisher/tests/test_views.py [hunk 1]
```diff
@@ -1548,6 +1548,41 @@ def test_details_page_with_course_runs_lms_id(self):
         response = self.client.get(self.detail_page_url)
         self.assertContains(response, 'course/{}'.format(lms_course_id))
 
+    def test_page_enable_waffle_switch_pilot(self):
+        """ Verify that user will see only studio fields when 'publisher_hide_features_for_pilot' is activated. """
+        self.user.groups.add(Group.objects.get(name=INTERNAL_USER_GROUP_NAME))
+        toggle_switch('publisher_hide_features_for_pilot', True)
+        response = self.client.get(self.detail_page_url)
+
+        self.assertContains(response, '<div id="approval-widget" class="hidden">')
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_views.py:1554]: you are checking for `approval-widget`  but docstring suggest that this test is for studio fields. ?

File: course_discovery/apps/publisher/tests/test_views.py [hunk 2]
```diff
@@ -1548,6 +1548,41 @@ def test_details_page_with_course_runs_lms_id(self):
...
+
+    def test_page_disable_waffle_switch_pilot(self):
+        """ Verify that user will see whole page when 'publisher_hide_features_for_pilot' is deactivated. """
+        self.user.groups.add(Group.objects.get(name=INTERNAL_USER_GROUP_NAME))
+        toggle_switch('publisher_hide_features_for_pilot', False)
+        response = self.client.get(self.detail_page_url)
+
+        self.assertContains(response, '<div id="approval-widget" class="">')
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_views.py:1562]: same as above

File: course_discovery/apps/publisher_comments/forms.py [hunk 3]
```diff
@@ -18,6 +18,10 @@ def get_comment_create_data(self):
         data['modified'] = self.cleaned_data['modified']
         return data
 
+    def __init__(self, *args, **kwargs):
+        super(CommentsForm, self).__init__(*args, **kwargs)
+        self.fields['comment'].widget.attrs['rows'] = 4
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher_comments/forms.py:23]: why you are not adding this using CSS directly?

File: course_discovery/static/sass/publisher/course_form.scss [hunk 4]
```diff
@@ -53,6 +53,7 @@
   border: 2px solid #d3d3d3;
   padding-top: 2px;
   padding-bottom: 2px;
+  padding-left: 3px;
```
Comment (tasawernawaz) [inline course_discovery/static/sass/publisher/course_form.scss]: i18n

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher_comments/forms.py (hunk 3 above)
Comment (awais786) [reply course_discovery/apps/publisher_comments/forms.py:23]: Django providing a way to fix text area size.

**Comment by awais786** [comment]
screen shot 2017-02-10 at 4 17 36 pm

File: course_discovery/apps/publisher/tests/test_views.py (hunk 1 above)
Comment (awais786) [reply course_discovery/apps/publisher/tests/test_views.py:1554]: Done

File: course_discovery/apps/publisher/tests/test_views.py (hunk 2 above)
Comment (awais786) [reply course_discovery/apps/publisher/tests/test_views.py:1562]: Done

**Review by tasawernawaz — APPROVED** [review]

**Comment by awais786** [comment]
@bderusha Please review.

**Comment by awais786** [comment]
@bderusha Please review.

**Comment by awais786** [comment]
screen shot 2017-02-10 at 8 01 21 pm

File: course_discovery/apps/publisher/tests/test_views.py [hunk 5]
```diff
@@ -1548,6 +1548,41 @@ def test_details_page_with_course_runs_lms_id(self):
...
+        """
+        self.user.groups.add(Group.objects.get(name=INTERNAL_USER_GROUP_NAME))
+        toggle_switch('publisher_comment_widget_feature', True)
+        response = self.client.get(self.detail_page_url)
+
+        self.assertContains(response, '<div id="comments-widget" class="comment-container ">')
+
+    def test_comments_with_disable_switch(self):
+        """ Verify that user will not see the comments widget when
+        'publisher_comment_widget_feature' is disable.
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_views.py]: `is disabled`

**Review by bderusha — APPROVED** [review]

**Pull request merged**